### PR TITLE
Iss90/get route window

### LINKED ISSUE #90: Filter route fixes to next on route fix given aircraft position
As described in the comment in `Airspace._get_route_window(...)`

https://github.com/project-bluebird/BluebirdATC/blob/807c64df24283622ba475399faa7d58fa1455078/bluebird-dt/bluebird_dt/core/airspace.py#L819

### PR DESCRIPTION
**Pull request by dlpbc**
Fixes #90

### CODE REVIEW

File: bluebird-dt/bluebird_dt/core/airspace.py [hunk 1]
```diff
@@ -814,17 +814,19 @@ def _get_route_windows(
...
         # find out which fixes on the route are in or out of the sector laterally
         # do not use any buffer around boundary - it's fine if fixes aren't exactly on it
         point_in_sector = [airspace.contains_laterally(location, epsilon=0) for location in point_locations]
 
-        # name the initial  position of the aircraft in the route
-        route = ["START_LOC", *route]
+        # if the first fix in the route is in the sector (which implies the route start within the sector)
+        # and the aircraft is on route, and its next fix is the first fix, then insert the aircraft position
+        # at the start of the route.
+        if point_in_sector[0] and aircraft.on_route and self.closest_forward_fix(aircraft, 0.0, route) == route[0]:
```
Comment (AlvaroSierra) [inline bluebird-dt/bluebird_dt/core/airspace.py]: Why would coordination logic depend on the tactical state of the aircraft?

**Comment by AlvaroSierra** [comment]
Thank you Ese, I don't really understand the motive for this change, but I might just be missing the context.

**Review by nbarlowATI — COMMENTED** [review]
It's not quite clear to me how this change filters the route fixes based on the aircraft location - won't `route` always be the same, regardless of where the aircraft is?  Or probably I'm just not quite understanding it.    Maybe a test that would demonstrate the new behaviour would be useful?

File: bluebird-dt/bluebird_dt/core/airspace.py (hunk 1 above)
Comment (dlpbc) [reply bluebird-dt/bluebird_dt/core/airspace.py]: I'm not sure what you mean by coordination logic in this context.

The idea is to ensure that the returned route windows/legs cover the defined route, and possibly an "artificial leg" defined based on the aircraft position and first fix in the route if the aircraft is yet to reach that fix (e.g., it was spawned at a position before the first fix).

**Comment by dlpbc** [comment]
> Thank you Ese, I don't really understand the motive for this change, but I might just be missing the context.

The current implementation inserts the aircraft’s current position at the start of the route regardless of whether it has flown past the first fix or whether the first fix is in the sector. This leads to a bug as the insertion is incorrect if the aircraft has flown past the first fix. Furthermore, if the defined route crosses the sector/airspace (into and out of the sector), inserting the aircraft’s position in the route is unnecessary.

This fix inserts the aircraft’s position at the start of the route only when it is necessary. That is, the first fix is in the sector/airspace and the aircraft has not yet flown past it. In this scenario, an "artificial leg" defined based on the aircraft position and first fix in the route is constructed and added to the list of route windows to return. I believe this was the intent of the original code that contains the bug.

**Comment by dlpbc** [comment]
> It's not quite clear to me how this change filters the route fixes based on the aircraft location - won't `route` always be the same, regardless of where the aircraft is? Or probably I'm just not quite understanding it. Maybe a test that would demonstrate the new behaviour would be useful?

The changes address a specific edge case and ensure that the edge-case logic is not used when it is not needed (this is the current implementation which produces a bug).

The edge case addresses scenarios where ah aircraft is yet to fly past the first fix in the route and the route starts inside the sector. An artificial leg/window is constructed from the aircraft's position to the first fix. This idea exists already in the current codebase but it contains a bug as it's used generally rather than for the specific edge case.

I'll create a test to demonstrate the issue.

**Review by AlvaroSierra — APPROVED** [review]

**Review by nbarlowATI — APPROVED** [review]

**Comment by dlpbc** [comment]
> > It's not quite clear to me how this change filters the route fixes based on the aircraft location - won't `route` always be the same, regardless of where the aircraft is? Or probably I'm just not quite understanding it. Maybe a test that would demonstrate the new behaviour would be useful?
> 
> The changes address a specific edge case and ensure that the edge-case logic is not used when it is not needed (this is the current implementation which produces a bug).
> 
> The edge case addresses scenarios where ah aircraft is yet to fly past the first fix in the route and the route starts inside the sector. An artificial leg/window is constructed from the aircraft's position to the first fix. This idea exists already in the current codebase but it contains a bug as it's used generally rather than for the specific edge case.
> 
> I'll create a test to demonstrate the issue.

Thanks @AlvaroSierra and @nbarlowATI for the review/approval. There's no need for the test/demo anymore.
